File: LeetCode Problems/1 to 100/55. Jump Game/55. Jump Game.cpp
class Solution {
public:
    bool canJump(vector<int> &nums) {
        int max = 0;
        for (int i = 0; i <= max; i++) {
            max = std::max(i + nums[i], max);
            if (max >= nums.size() - 1)
                return true;
        }
        return false;
    }
};

File: LeetCode Problems/1 to 100/55. Jump Game/55. Jump Game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int indx = n - 1;
        for (int i = n - 2; i >= 0; i--) {
            if (i + nums[i] >= indx) indx = i;
        }
        return indx == 0;
    }
}
